From 10338c8ef0c876a2deb414e2a00876246b49b6f4 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 24 Aug 2022 20:25:40 +0300
Subject: [PATCH] [SW-86869] common: check image pointer size for max value

Currently we read image pointer (manifest) size from 0x60000 SPI offset
for SPI manifest and 0x0 offset for EROM manifest and performs the DMA
of the manifest image. The minimum size of manifest image is checked inside
DMA function but not the maximum size. This patch adds max image
size check for manifest image which is 64KB.

Change-Id: Ie8158dfc9e9ebe3453ad8eb41e450829650c3f39
---

diff --git a/arch/gaudi3/arc/progress_bits.h b/arch/gaudi3/arc/progress_bits.h
index 89e2dd5..487c10d 100644
--- a/arch/gaudi3/arc/progress_bits.h
+++ b/arch/gaudi3/arc/progress_bits.h
@@ -7,140 +7,142 @@
 	PROGRESS_INIT = 0,
 	PROGRESS_INIT_SP_SRAM_NOT_COMPLETED = 1,
 	PROGRESS_INIT_ARC1_ICCM_INIT_FAILED = 2,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 3,
-	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 4,
-	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 5,
-	PROGRESS_INIT_SPI_BOOT_DESC_INVALID_IMG_SIZE = 6,
-	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 7,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 8,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 9,
-	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 10,
-	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 11,
-	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 12,
-	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 13,
-	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 14,
-	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 15,
-	PROGRESS_INIT_SPI_BOOT_DESC_FAIL = 16,
-	PROGRESS_INIT_EFUSE_SPI = 17,
-	PROGRESS_INIT_EFUSE_SPI_SLV_TIMEDOUT = 18,
-	PROGRESS_REVOKE_LOC = 19,
-	PROGRESS_REVOKE_LOC_0 = 20,
-	PROGRESS_REVOKE_LOC_1 = 21,
-	PROGRESS_KEY_LOC = 22,
-	PROGRESS_KEY_LOC_0 = 23,
-	PROGRESS_KEY_LOC_1 = 24,
-	PROGRESS_VER_LOC = 25,
-	PROGRESS_VER_LOC_0 = 26,
-	PROGRESS_VER_LOC_1 = 27,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 28,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 29,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 30,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 31,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 32,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 33,
-	PROGRESS_BRINGUP_PCI_PRI = 34,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 35,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 36,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 37,
-	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 38,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 39,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 40,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 41,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 42,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 43,
-	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 44,
-	PROGRESS_BRINGUP_PCI_SEC = 45,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 46,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 47,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 48,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 49,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 50,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 51,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 52,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 53,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 54,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 55,
-	PROGRESS_ARCBOOT_IN_RECOVERY = 56,
-	PROGRESS_ARCBOOT_ROM_TPM = 57,
-	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 58,
-	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 59,
-	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 60,
-	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 61,
-	PROGRESS_BRINGUP_TPM_SPI = 62,
-	PROGRESS_BRINGUP_TPM_SPI_PRI = 63,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 64,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 65,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 66,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_INVALID_IMG_SIZE = 67,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 68,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 69,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 70,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 71,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 72,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 73,
-	PROGRESS_BRINGUP_TPM_SPI_SEC = 74,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 75,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 76,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 77,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_INVALID_IMG_SIZE = 78,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 79,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 80,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 81,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 82,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 83,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 84,
-	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 85,
-	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 86,
-	PROGRESS_PPBOOT_START = 87,
-	PROGRESS_PPBOOT_FINISH = 88,
-	PROGRESS_BRINGUP_RAM_PRI = 89,
-	PROGRESS_BRINGUP_RAM_PRI_DMA = 90,
-	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 91,
-	PROGRESS_BRINGUP_RAM_SEC = 92,
-	PROGRESS_BRINGUP_RAM_SEC_DMA = 93,
-	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 94,
-	PROGRESS_ARCBOOT_RAM_START = 95,
-	PROGRESS_ARCBOOT_RECOVERY = 96,
-	PROGRESS_ARCBOOT_RECOVERY_DMA = 97,
-	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 98,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 99,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 100,
-	PROGRESS_ARCBOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 101,
-	PROGRESS_ARCBOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 102,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 103,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 104,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 105,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 106,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 107,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 108,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 109,
-	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 110,
-	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 111,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 112,
-	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 113,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 114,
-	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_SIZE = 115,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 116,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 117,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 118,
-	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 119,
-	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 120,
-	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 121,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 122,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 123,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 124,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 125,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 126,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 127,
-	PROGRESS_ARCBOOT_EROM_IMG0 = 128,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 129,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 130,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 131,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_INVALID_IMG_SIZE = 132,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 133,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 134,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 135,
-	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 136,
-	PROGRESS_ARCBOOT_EROM_IMG0_START = 137,
-	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 138,
+	PROGRESS_INIT_SPI_BOOT_DESC_EXCEEDS_MAX_SIZE = 3,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 4,
+	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 5,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 6,
+	PROGRESS_INIT_SPI_BOOT_DESC_INVALID_IMG_SIZE = 7,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 8,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 9,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 10,
+	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 11,
+	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 12,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 13,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 14,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 15,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 16,
+	PROGRESS_INIT_SPI_BOOT_DESC_FAIL = 17,
+	PROGRESS_INIT_EFUSE_SPI = 18,
+	PROGRESS_INIT_EFUSE_SPI_SLV_TIMEDOUT = 19,
+	PROGRESS_REVOKE_LOC = 20,
+	PROGRESS_REVOKE_LOC_0 = 21,
+	PROGRESS_REVOKE_LOC_1 = 22,
+	PROGRESS_KEY_LOC = 23,
+	PROGRESS_KEY_LOC_0 = 24,
+	PROGRESS_KEY_LOC_1 = 25,
+	PROGRESS_VER_LOC = 26,
+	PROGRESS_VER_LOC_0 = 27,
+	PROGRESS_VER_LOC_1 = 28,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 29,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 30,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 31,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 32,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 33,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 34,
+	PROGRESS_BRINGUP_PCI_PRI = 35,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 36,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 37,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 38,
+	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 39,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 40,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 41,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 42,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 43,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 44,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 45,
+	PROGRESS_BRINGUP_PCI_SEC = 46,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 47,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 48,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 49,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 50,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 51,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 52,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 53,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 54,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 55,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 56,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 57,
+	PROGRESS_ARCBOOT_ROM_TPM = 58,
+	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 59,
+	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 60,
+	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 61,
+	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 62,
+	PROGRESS_BRINGUP_TPM_SPI = 63,
+	PROGRESS_BRINGUP_TPM_SPI_PRI = 64,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 65,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 66,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 67,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_INVALID_IMG_SIZE = 68,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 69,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 70,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 72,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 72,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 73,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 74,
+	PROGRESS_BRINGUP_TPM_SPI_SEC = 75,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 76,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 77,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 78,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_INVALID_IMG_SIZE = 79,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 80,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 81,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 82,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 83,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 84,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 85,
+	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 86,
+	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 87,
+	PROGRESS_PPBOOT_START = 88,
+	PROGRESS_PPBOOT_FINISH = 89,
+	PROGRESS_BRINGUP_RAM_PRI = 90,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 91,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 92,
+	PROGRESS_BRINGUP_RAM_SEC = 93,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 94,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 95,
+	PROGRESS_ARCBOOT_RAM_START = 96,
+	PROGRESS_ARCBOOT_RECOVERY = 97,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 98,
+	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 99,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 100,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 101,
+	PROGRESS_ARCBOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 102,
+	PROGRESS_ARCBOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 103,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 104,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 105,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 106,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 107,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 108,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 109,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 110,
+	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 111,
+	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 112,
+	PROGRESS_INIT_EROM_BOOT_DESC_EXCEEDS_MAX_SIZE = 113,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 114,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 115,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 116,
+	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_SIZE = 117,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 118,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 119,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 120,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 121,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 122,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 123,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 124,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 125,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 126,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 127,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 128,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 129,
+	PROGRESS_ARCBOOT_EROM_IMG0 = 130,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 131,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 132,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 133,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_INVALID_IMG_SIZE = 134,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 135,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 136,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 137,
+	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 138,
+	PROGRESS_ARCBOOT_EROM_IMG0_START = 139,
+	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 140,
 };
diff --git a/common/boot_desc.c b/common/boot_desc.c
index acb8cc4..885f13d 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -95,6 +95,11 @@
 	}
 
 	img_ptrs_size = spi_flash_readl(p_img_ptr_ctx->img_ptr_spi_offset);
+	if (img_ptrs_size > HL_IMG_PTRS_MAX_SIZE) {
+		arcboot_rom_status_update_bringup(dma_sts_base,
+				PROGRESS_BOOT_DESC_EXCEEDS_MAX_SIZE);
+		return 0;
+	}
 
 	arcboot_rom_status_update_bringup(dma_sts_base, PROGRESS_DMA_START);
 
diff --git a/common/boot_desc.h b/common/boot_desc.h
index dbf0c01..0d0ccc6 100644
--- a/common/boot_desc.h
+++ b/common/boot_desc.h
@@ -42,6 +42,8 @@
 					HL_SPI_BOOT_PTR_SP_SRAM_OFF)
 #define HL_BOOT_PTS_COUNT_FLASH_OFFSET	(BOOT_PTRS_SZ - sizeof(uint32_t))
 
+#define HL_IMG_PTRS_MAX_SIZE		0x10000 /* 64KB */
+
 struct boot_desc {
 	uint32_t desc_ver;
 	uint32_t spi_offset;
diff --git a/include/arcboot.h b/include/arcboot.h
index 381a79b..a14ccad 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -108,6 +108,7 @@
 };
 
 enum arcboot_rom_boot_desc_progress {
+	PROGRESS_BOOT_DESC_EXCEEDS_MAX_SIZE,
 	PROGRESS_DMA_START,
 	PROGRESS_DMA_CRC_FAILED,
 	PROGRESS_DMA_UNALIGNED_DEST,
